### ISSUE TITLE
Base image version (debian stretch, ubuntu bionic)

### ISSUE THREAD

**Issue by andriy-f** [docker-library/mongo#313]
Hi, just curious
1. Did you have any reasons not to use Debian Stretch or Ubuntu bionic (18 LTS) for MongoDB 4?

2. In Dockerfile you sometimes use

```sh
set -ex \
  command1; \
  command2; \
```

and sometimes

```sh
set -x \
  command1 &&
  command2 
```

what's the benefit of one over other if applicable?

**Comment by andriy-f** [comment]
https://github.com/docker-library/mongo/pull/275
https://github.com/docker-library/mongo/pull/297

**Comment by yosifkit** [comment]
We did try to update to bionic (https://github.com/docker-library/mongo/pull/293) but lost architectures and so reverted it with #297.  We moved to Ubuntu from Debian based images to get more architecture support.

As for the reason for ampersand versus semicolon on the `RUN` line, shells have unwanted behavior with something like a `for` loop within a complex expression (`&&` chain) but the rest of the time it is basically equivalent.  As specific parts are updated, we've been slowly moving toward the semicolon version with `set -e`, but it is not worth rebuilding the image for what is mostly a stylistic change.
